### PR TITLE
FLUID-4059: Update demos to current idioms

### PR DESCRIPTION
**Pull request by jobara**
https://issues.fluidproject.org/browse/FLUID-4059

### CODE REVIEW

File: demos/progress/js/progressDemo.js [hunk 1]
```diff
@@ -18,135 +18,169 @@ var demo = demo || {};
...
+    };
+
     /**
      * Used to simulate an application that would call Progress.
-     * @param {Integer} percent         the starting percentage when first called and then the current percentage when called recursively
-     * @param {Integer} steps           the number of recursive steps to use for the simulated progress
-     *                                  note: because of some randomness inserted into the simulation
-     *                                  for realism, the number of steps will actually be much less
+     * @param {Object} that             the component to source the model and applier from
      * @param {Function} stepFunction   the stepFunction update the progress component
```
Comment (cindyli) [inline demos/progress/js/progressDemo.js]: change "update" -> "to update"?

File: demos/progress/js/progressDemo.js [hunk 2]
```diff
@@ -18,135 +18,169 @@ var demo = demo || {};
...
      * @param {Function} finishFunction the finishFunction enables the submit button, hides the progress simulation and update the text
```
Comment (cindyli) [inline demos/progress/js/progressDemo.js]: Change "enables" -> "to enable", "hides" -> "hide"?

File: demos/progress/js/progressDemo.js [hunk 3]
```diff
@@ -18,135 +18,169 @@ var demo = demo || {};
...
+            "onCreate.setAria": {
+                funcName: "demo.shoppingDemo.setAriaAttr",
+                args: ["{that}.dom.liveRegion", "{that}.dom.submitButton", "{that}.containerID"]
+            },
+            "onCreate.bindSubmitEvents": {
+                "this": "{that}.dom.submitButton",
+                "method": "on",
+                args: ["click", "{that}.events.onSubmit.fire"]
+            },
+            "onSubmit.": {
```
Comment (cindyli) [inline demos/progress/js/progressDemo.js]: Missing namespace.

File: demos/progress/js/progressDemo.js [hunk 4]
```diff
@@ -18,135 +18,169 @@ var demo = demo || {};
...
+                    args: ["{that}.container"]
+                }
+            }
+        },
+        components: {
+            progress: {
+                type: "fluid.progress",
+                container: "{that}.dom.progress",
+                options: {
+                    speed: 1000,
```
Comment (cindyli) [inline demos/progress/js/progressDemo.js]: Would be nice to have this options configurable at the top component level `demo.shoppingDemo`. Also, can you add a code comment like `animation speed in milliseconds` etc?

Same comment applies to line 138, 143.

File: demos/progress/js/progressDemo.js [hunk 5]
```diff
@@ -18,135 +18,169 @@ var demo = demo || {};
...
+                            excludeSource: "init"
+                        }
+                    }
+                }
+            }
+        }
+    });
 
-    var setAriaAttr = function (liveRegion, submitButton) {
+    demo.shoppingDemo.setAriaAttr = function (liveRegion, submitButton, controlsID) {
```
Comment (cindyli) [inline demos/progress/js/progressDemo.js]: These three functions can be replaced by a declarative way.

File: demos/renderer/js/rendererDemo.js [hunk 6]
```diff
@@ -12,226 +12,219 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
 /* global fluid */
 
 var demo = demo || {};
-(function ($) {
+(function () {
```
Comment (cindyli) [inline demos/renderer/js/rendererDemo.js]: Fix the copyright.

**Comment by jobara** [comment]
@cindyli made the changes you suggested. Ready for more review.

**Comment by cindyli** [comment]
Merged at c8457da92655335c042316b497a8d0bfb1f7c9fc

**Pull request merged**